scapy/layers/dns.py

```diff
@@ -88,6 +88,30 @@ def dns_get_str(s, p, pkt=None, _internal=False):
     return name, p, s[burned:]
 
 
+def dns_encode(x, check_built=False):
+    """Encodes a bytes string into the DNS format
+
+    :param x: the string
+    :param check_built: detect already-built strings and ignore them
+    :returns: the encoded bytes string
+    """
+    if not x or x == b".":
+        return b"\x00"
+
+    built = check_built
+    built &= b"." not in x
```

Comment:
I'm personally not a fan of `&=`. I would have written this as
```
if check_built:
    if b"." not in x and \
       ((x[-1] == b"\x00") or ((orb(x[-2]) & 0xc0)) == 0xc0):
        return x
```

Reply:
Agreed (because `&` is for bitwise operations). But there is no need to use two `if` statements.

I think this is easier to read (there are unnecessary parenthesis in the original code):
```Python
    if check_built and b"." not in x and (
            x[-1] == b"\x00" or orb(x[-2]) & 0xc0 == 0xc0
    ):
```

Reply:
Updated thanks !